pytorch_lightning/core/lightning.py

```diff
@@ -91,11 +91,53 @@ def forward(self, x):
         if self.trainer.proc_rank == 0:
             print(*args, **kwargs)
 
+    # Note this is almost identical to distrib_parts.TrainerDPMixin.__transfer_data_to_device
```

Comment:
I thought about this, I didn't want to import across because that would be worse.

I think the best way to go about this would probably be to move data transfers into the utils sections so that they can be used by other stuff

Reply:
so can we use `distrib_parts.TrainerDPMixin.__transfer_data_to_device` with some level of generalization to serve for both cases?

Reply:
Yep, I've already made the change to deal with that @Borda